config/routes.rb

```diff
@@ -1,7 +1,9 @@
-# frozen_string_literal: true
-
 Rails.application.routes.draw do
-  root 'home#index'
+  root 'users#index'
+  get 'users/:user_id' => 'users#show', :defaults => { format: 'html' }, as: :user
+  get '/users/:user_id/posts/:post_id', to: 'users#user_post', as: 'user_post'
+  get '/users/:user_id/posts', to: 'users#user_posts', as: 'user_posts'
```

Comment:
- [ ] [OPTIONAL] Kindly implement the routes like the following as this is a cleaner way of writing it and should be easier to modify in the future.

```ruby 
resources :users, only: [:index, :show] do
   resources :posts, only: [:index, :show]
end

root 'users#index'
```